src/NovelRT.Interop/Animation/NovelRTSpriteAnimatorState.cpp

```diff
@@ -0,0 +1,201 @@
+    
+    *outputLoop = static_cast<int32_t>(cppState->shouldLoop());
+    
+    return NOVELRT_SUCCESS;
+}
+
+int32_t NovelRT_SpriteAnimatorState_setShouldLoop(NovelRTSpriteAnimatorState state, int32_t loop, const char** errorMessage) {
+    if (state == nullptr) {
+        if (errorMessage != nullptr) {
+            *errorMessage = NovelRT_getErrMsgIsNullptr();
+        }
+        return NOVELRT_FAILURE;
+    }
+
+    Animation::SpriteAnimatorState* cppState = reinterpret_cast<Animation::SpriteAnimatorState*>(state);
+    
+    cppState->shouldLoop() = static_cast<bool>(loop);
+    
+    return NOVELRT_SUCCESS;
+}
+
+int32_t NovelRT_SpriteAnimatorState_getFrames(NovelRTSpriteAnimatorState state, NovelRTSpriteAnimatorFrameVector* outputFrames, const char** errorMessage) {
+    if (state == nullptr || outputFrames == nullptr) {
+        if (errorMessage != nullptr) {
+            *errorMessage = NovelRT_getErrMsgIsNullptr();
+        }
+        return NOVELRT_FAILURE;
+    }
+
+    Animation::SpriteAnimatorState* cppState = reinterpret_cast<Animation::SpriteAnimatorState*>(state);
+    std::vector<Animation::SpriteAnimatorFrame>* frames = new std::vector<Animation::SpriteAnimatorFrame>();
+    *frames = cppState->frames();
+
+    *outputFrames = reinterpret_cast<NovelRTSpriteAnimatorFrameVector>(&frames);
+    return NOVELRT_SUCCESS;
+}
+
+int32_t NovelRT_SpriteAnimatorState_setFrames(NovelRTSpriteAnimatorState state, NovelRTSpriteAnimatorFrameVector frames, const char** errorMessage) {
+    if (state == nullptr || frames == nullptr) {
+        if (errorMessage != nullptr) {
+            *errorMessage = NovelRT_getErrMsgIsNullptr();
+        }
+        return NOVELRT_FAILURE;
+    }
+
+    Animation::SpriteAnimatorState* cppState = reinterpret_cast<Animation::SpriteAnimatorState*>(state);
+    std::vector<Animation::SpriteAnimatorFrame>* cppFrames = reinterpret_cast<std::vector<Animation::SpriteAnimatorFrame>*>(frames);//new std::vector<Animation::SpriteAnimatorFrame>();
+    cppState->frames() = *cppFrames;
+    return NOVELRT_SUCCESS;
+}
+
+int32_t NovelRT_SpriteAnimatorState_tryFindValidTransition(NovelRTSpriteAnimatorState state, NovelRTSpriteAnimatorState* outputTransitionState, const char** errorMessage) {
+    if (state == nullptr || outputTransitionState == nullptr) {
+        if (errorMessage != nullptr) {
+            *errorMessage = NovelRT_getErrMsgIsNullptr();
+        }
+        return NOVELRT_FAILURE;
+    }
+
+    Animation::SpriteAnimatorState* cppState = reinterpret_cast<Animation::SpriteAnimatorState*>(state);
+    
+    try {
+        *outputTransitionState = reinterpret_cast<NovelRTSpriteAnimatorState>(cppState->tryFindValidTransition().get());
+    }
+    catch (const std::exception& ex) {
+        if (errorMessage != nullptr) {
+            *errorMessage = ex.what();
+        }
+        return NOVELRT_FAILURE;
+    }
+    
+    return NOVELRT_SUCCESS;    
+}
+
+//std::vector<SpriteAnimatorFrame>
```

Comment:
Dead code?

Reply:
Just a comment to help separate types - Removed.